tests/test_api.py

```diff
@@ -79,3 +82,26 @@ def test_collection_argument_with_dollar() -> None:
                 matched = True
                 assert action.ctl.ruleRemoveTargetById == 942290
                 assert action.ctl.removeVariableName == "json.flags.$notjunk"
+
+def test_lowercase_and_uppercase_in_argument() -> None:
+    """Test that a collection argument can contain `$` (e.g., a key in a JSON document)"""
+    rule_text = """
+    SecRule REQUEST_FILENAME "@rx /[ABCD]+/$" \
+    "id:1234,\
+    phase:1,\
+    pass,\
+    t:lowercase,\
+    nolog
+    """
+
+    uppercase_regex = re.compile('/[A-Z]/')
+    parsed_rule = parser.process_from_str(rule_text)
+    for rule in parsed_rule.rules:
+        assert rule.__class__.__name__ == "SecRule"
+        for action in rule.actions:
+            pprint.pprint(action)
+            if action.transformations:
+                for t in action.transformations:
+                    if t == "lowercase":
+                        pprint.pprint(rule)
+                        assert not uppercase_regex.match(rule.operator.rx)
```

Comment:
You should also check a "matched" variable at the end, like I intended to in my test. Otherwise, because of the loops, it is not guaranteed that the assertions are actually executed.